superset/connectors/sqla/utils.py

```diff
@@ -180,6 +180,15 @@ def is_column_type_temporal(column_type: TypeEngine) -> bool:
         return False
 
 
+def convert_column_type(column_type: TypeEngine) -> str:
+    if column_type.python_type == list:
+        return "ARRAY"
+    if column_type.python_type == dict:
+        return "JSON"
```

Comment:
Can it also be a `STRUCT`? I guess it probably depends on the database.

Reply:
so i'm following the pattern from the `sqlalchemy.sql.sqltypes`

Reply:
> Should we save the original database type here? The type field specifically used Text as storage so the originally intention is probably to allow storing the full schema of complex types such as ARRAY and STRUCT?

I agree, using the original native type is the ideal solution here.